### DISCUSSION TITLE
Passing request json through Azure function

### DISCUSSION THREAD

**Discussion by mtelang** [microsoft/RulesEngine#428] [Q&A]
Hello @abbasc52 

First, thank you for sharing this outstanding tool. 

I have created an Azure function 

namespace HttpRuleFunction
{
    public static class HttpRuleFunction     {
        [FunctionName("HttpRuleFunction")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
            ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");

            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
            dynamic data = JsonConvert.DeserializeObject(requestBody);

            string discountOffered = new DemoApp.JSONDemo().Run(data);

            string responseMessage = string.IsNullOrEmpty(discountOffered)
                ? "Something went wrong. discountOffered string was empty."
                : $"{discountOffered}.";

            return new OkObjectResult(responseMessage);
        }
    }
}

The http request body is 

**{"country" : "india",
"loyaltyFactor" : 5}**

Also, I have simplified the Discount.json
**[
  {
    "WorkflowName": "Discount",
    "Rules": [
      {
        "RuleName": "GiveDiscount10",
        "SuccessEvent": "10",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "propertyExpando.country == \"india\" AND propertyExpando.loyaltyFactor <= 2"
      },
      {
        "RuleName": "GiveDiscount20",
        "SuccessEvent": "20",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "propertyExpando.country == \"india\" AND propertyExpando.loyaltyFactor == 3"
      },
      {
        "RuleName": "GiveDiscount25",
        "SuccessEvent": "25",
        "ErrorMessage": "One or more adjust rules failed.",
        "ErrorType": "Error",
        "RuleExpressionType": "LambdaExpression",
        "Expression": "propertyExpando.country == \"india\" AND propertyExpando.loyaltyFactor == 5"
      }
    ]
  }
]**

But in the ResultSet, I am getting exception 

exceptionMessage	"Exception while parsing expression `propertyExpando.country == \"india\" AND propertyExpando.loyaltyFactor <= 2` - The binary operator LessThanOrEqual is not defined for the types 'System.Object' and 'System.Int32'."	

If I run default program passing var basicInfo = "{\"country\": \"india\",\"loyaltyFactor\": 5}"; it works fine.

Not sure why it is not accepting my JSON. Any help is appreciated

Thanks

**Comment by DavidKarolinskySoftacom** [comment]
I think that during the object journey through the function some serialization settings could be missed, a had the same issue with the nullable decimals and solved it in two ways: explicit convertation - `propertyExpando.country == \"india\" AND Int32(propertyExpando.loyaltyFactor)<= 2` or custom ConvertUtiity with static method to convert objects/JTokens to Int32.
